HHS/Head-Start-TTADP
# [Prod] Allow creator/approvers to reset their reports to draft, add training report scope to the backend

## Description of change

### TTAHUB-2721
Per user request - in the situation where a user is accidentally an approver and the creator of an activity report, and the report is in "needs action", allow that user to reset the report to draft.

### TTAHUB-2645
Add the "filter by goal name" scope to the backend for training reports. This is backend only and is one of 4 that need to be in place before the frontend can be complete.

## How to test
To test 2721: set yourself as an approver and creators on a report and then reset that report to draft. Confirm that the functionality doesn't exist for approvers who aren't creators, or creators who aren't approvers (on a given report)

2645 is a backend only change and is invisible to the user base

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2645
* https://jira.acf.gov/browse/TTAHUB-2721


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
